functional-tests/fn_bridge_deposit_happy.py

```diff
@@ -25,133 +26,163 @@ def __init__(self, ctx: flexitest.InitContext):
+        return txid
 
-        # Third deposit, address 1
-        # Create the first deposit request transaction
-        tx = deposit_request_transaction(el_address_1, btc_url, btc_user, btc_password)
-        self.logger.debug(f"Deposit request tx: {tx}")
-        txid = btcrpc.sendrawtransaction(tx)
-        original_balance_3 = int(rethrpc.eth_getBalance(f"0x{el_address_1}"), 16)
-        self.logger.debug(f"Balance before deposit: {original_balance_3}")
-        txid = btcrpc.sendrawtransaction(tx)
-        self.logger.debug(f"sent deposit request with txid = {txid} for address {el_address_1}")
-
-        # Now poll for the tx in chain
-        tx_published = False
-        for _ in range(10):
-            time.sleep(1)
-            try:
-                _ = btcrpc.gettransaction(txid)
-                print("Found expected tx in mempool")
-                tx_published = True
-                break
-            except Exception as e:
-                print(e)
-        assert tx_published, "Tx was not published"
-        # time to mature FIXME: is this 6 blocks?
-        btcrpc.proxy.generatetoaddress(6, seq_addr)
-        time.sleep(1)
+    def test_deposit(
+        self, ctx: flexitest.RunContext, address: str, el_address: str, new_address=True
+    ):
+        """
+        Test depositing funds into the bridge and verifying the corresponding increase in balance
+        on the Strata side.
+        """
+        rollup_deposit_amount = DEFAULT_ROLLUP_PARAMS["deposit_amount"]
+
+        btc = ctx.get_service("bitcoin")
+        seq = ctx.get_service("sequencer")
+        reth = ctx.get_service("reth")
+
+        self.logger.debug(f"EL address: {el_address}")
 
-        new_balance_3 = int(rethrpc.eth_getBalance(f"0x{el_address_1}"), 16)
-        assert new_balance_3 > original_balance_3, "balance did not increase"
+        seqrpc = seq.create_rpc()
+        btcrpc: BitcoindClient = btc.create_rpc()
+        rethrpc = reth.create_rpc()
+
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+
+        self.logger.debug(f"BTC URL: {btc_url}")
+        self.logger.debug(f"BTC user: {btc_user}")
+        self.logger.debug(f"BTC password: {btc_password}")
+
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        self.logger.debug(f"Bridge pubkey: {bridge_pk}")
+
+        seq_addr = seq.get_prop("address")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        self.logger.debug(f"Address: {address}")
+
+        # Make sure that the el_address has zero balance
+        original_balance = int(rethrpc.eth_getBalance(f"0x{el_address}"), 16)
+        self.logger.debug(f"Balance before deposit (EL address): {original_balance}")
+        if new_address:
+            assert original_balance == 0, "balance is not zero"
+        else:
+            assert original_balance > 0, "balance is zero"
```

Comment:
This is not strictly necessary. For the purposes of this test, we only care whether the deposit works i.e, whether the balance for a particular account increases by a certain amount.

Reply:
It would say it's fine an thorough to check that other accounts that we know we didn't deposit to are *not* credited.

Reply:
> I would say it's fine an thorough to check that other accounts that we know we didn't deposit to are not credited.

Well, if we _were_ making that assertion, sure. All this asserts right now is that some random address was not credited, and not that no other account was erroneously credited.

Reply:
https://alpenlabs.atlassian.net/browse/STR-541